BackofficeBundle/Controller/ContentTypeController.php

```diff
@@ -77,22 +77,14 @@ public function newAction(Request $request)
         );
 
         $form->handleRequest($request);
+        if (!$request->get('no_save')) {
+            if ($this->handleForm($form, $this->get('translator')->trans('open_orchestra_backoffice.form.content_type.creation'), $contentType)) {
+                $this->dispatchEvent(ContentTypeEvents::CONTENT_TYPE_CREATE, new ContentTypeEvent($contentType));
 
-        if ($form->isValid()) {
-            $documentManager = $this->get('doctrine.odm.mongodb.document_manager');
-            $documentManager->persist($contentType);
-            $documentManager->flush();
-
-            $this->get('session')->getFlashBag()->add(
-                'success',
-                $this->get('translator')->trans('open_orchestra_backoffice.form.content_type.creation')
-            );
-
-            $this->dispatchEvent(ContentTypeEvents::CONTENT_TYPE_CREATE, new ContentTypeEvent($contentType));
-
-            return $this->redirect(
-                $this->generateUrl('open_orchestra_backoffice_content_type_form', array('contentTypeId' => $contentType->getContentTypeId()))
-            );
+                return $this->redirect($this->generateUrl('open_orchestra_backoffice_content_type_form', array(
```

Comment:
Befor the redirection, can you check if the contentType was valid and saved in the database (either you check the form, or if the mongo id ofthe contentType is set)